### DISCUSSION TITLE
Using composer across environments

### DISCUSSION THREAD

**Discussion by whitsey** [composer/composer#10721] [General]
I have just found composer and have installed a few packages.

My dev profile has: 
1. dev environment on local PC
2. Staging environment on remote server
3. Live environment on remote server
This results in 3 versions/installs of composer

Currently I run `composer require xxx/xxx` on local.
Then I publish and run again in staging
Then I publish and run again in live

Effectively maintaining 3 separate installs of composer.

Is this the recommended way of using composer across environments? 
Help on this topic seems to be difficult to understand

**Comment by stof** [comment]
On other environments, the deployment process should run `composer install` (with the lock file being committed). You should not re-run the `require` command on them.
